x-pack/plugins/alerting/server/alerts_client/alerts_client.ts

```diff
@@ -130,6 +132,8 @@ export class AlertsClient<
     this.rule = formatRule({ rule: this.options.rule, ruleType: this.options.ruleType });
     this.ruleType = options.ruleType;
     this._isUsingDataStreams = this.options.dataStreamAdapter.isUsingDataStreams();
+    this.ruleInfoMessage = `for ${this.ruleType.id}:${this.options.rule.id} '${this.options.rule.name}'`;
+    this.logTags = { tags: [this.ruleType.id, this.options.rule.id, 'alerts-client'] };
   }
```

Comment:
With this PR: https://github.com/elastic/kibana/pull/187785, the logger that's passed into the alerts client should already include the rule ID and rule type ID in the tags, so I think you just need to add the `alerts-client` tag.

Reply:
I tried this out, and it didn't add the tags.  Looks like we're getting a normal logger here
https://github.com/elastic/kibana/blob/385281b2a6c9f12531c7672d2cb56468af080b62/x-pack/plugins/alerting/server/task_runner/task_runner.ts#L171

which is then replaced in `run()` here: 
https://github.com/elastic/kibana/blob/385281b2a6c9f12531c7672d2cb56468af080b62/x-pack/plugins/alerting/server/task_runner/task_runner.ts#L670

Looking through the code, seems like it should have picked up ours, not the original one, not sure why it didn't.

I tested this by changing a condition around one of the comparisons in the code that would generate a log, like here: https://github.com/elastic/kibana/blob/385281b2a6c9f12531c7672d2cb56468af080b62/x-pack/plugins/alerting/server/alerts_client/alerts_client.ts#L530 

I changed `!isValidAlertIndexName()` to `isValidAlertIndexName()`.

To see the tags when running, I had to switch to JSON mode:

```yaml
logging:
  appenders:
    console_appender:
      type: console
      layout:
        type: json
#        type: pattern - what I normally use
#        highlight: true
```

I also ran this in a debugger and saw that it wasn't our logger but a base one, when the alertsClient logger logged.

I feel like opening another issue to investigate this, rather than trying to figure out why, then fix it, and find 10K tests to fix :-).  But maybe it's simpler?

Reply:
Weird. But I'm fine with figuring it out later :)

Reply:
Looks like it's here:
https://github.com/elastic/kibana/blob/07717a43ab369847d87c8e15071759502a89c48b/x-pack/plugins/alerting/server/alerts_service/alerts_service.ts#L223

Instead of passing in `this.options.logger` into the AlertsClient, we could pass in `opts.logger` to get the TaskRunner logger. But again, it's not a big deal and I don't think it's something we need to update rn.